docs/buildstrategies.md

```diff
@@ -412,6 +413,86 @@ Contrary to the strategy `spec.parameters`, you can use system parameters and th
 | System Parameter   |    No        |  At run-time, by the `BuildRun` controller.  |
 | Strategy Parameter |    Yes       |  At build-time, during the `BuildStrategy` creation. |
 
+## Securely referencing parameters
+
+In build strategy steps, parameters are referenced using `$(params.PARAM_NAME)`. This applies to system parameters, and those parameters defined in the build strategy. You can reference those parameters at many locations in the build steps, such as environment variables values, arguments, image, and more. In the Pod, all `$(params.PARAM_NAME)` tokens will be replaced by simple string replaces. This is safe in most locations but requires your attention when you define an inline script using an argument. For example:
```

Comment:
I think u need to make the distinction on this applying only to parameters of the type string, the parameters of the type array do not fall into this potential security attack vector right?

Reply:
That's because the only way to access arrays is through args:

```yaml
  args:
  - $(params.array-param[*])
```

This indeed is always safe. I'll update the wording.

Reply:
Done @qu1queee.